### PR TITLE
test(component): refactor Button component tests

### PR DESCRIPTION
**Pull request by bc-alexsaiannyi**
## What?

This PR is aimed to refactor Button component's tests according to best practices. We take into account recommendations from  Kent C. Dodds [article](https://kentcdodds.com/blog/common-mistakes-with-react-testing-library#using-container-to-query-for-elements). For instance, using `container` for  `query`-ing elements has been removed. Instead,  we replace it with `screen`. 

**NOTE:**   We've also switched to using `userEvent` instead of `fireEvent` whenever it's possible. For this purpose a package `@testing-library/user-event` has been added and `yarn.lock` is updated as well.
## Why?
the changes are based on GitHub [issue](https://github.com/bigcommerce/big-design/issues/635) and PRs will be open for each component separately.

## Testing/Proof

The tests have been successfully run locally.

### CODE REVIEW

**Comment by chanceaclark** [comment]
Hey @bc-alexsaiannyi if this is a draft PR, would you mind using the [Draft Pull Request feature](https://github.blog/2019-02-14-introducing-draft-pull-requests/) instead of appending `[DRAFT]`? It'll leave reviewers off until you are ready for review.

**Comment by bc-alexsaiannyi** [comment]
@chanceaclark , sorry. I will do it next time. This time you can check my changes)

File: packages/big-design/src/components/Button/spec.tsx [hunk 1]
```diff
@@ -9,180 +10,179 @@ import { StyleableButton } from './private';
...
 
 test('triggers onClick', () => {
   const onClick = jest.fn();
 
-  const { container } = render(<Button onClick={onClick} />);
-  const button = container.firstChild as HTMLElement;
+  render(<Button onClick={onClick} />);
 
-  fireEvent.click(button);
+  fireEvent.click(screen.getByRole<HTMLButtonElement>('button'));
```
Comment (chanceaclark) [inline packages/big-design/src/components/Button/spec.tsx:159]: Let's also switch out usages of `fireEvent` with `userEvent`.

File: packages/big-design/src/components/Button/spec.tsx [hunk 2]
```diff
@@ -9,180 +10,179 @@ import { StyleableButton } from './private';
...
-  const { getAllByTestId } = render(
+
+  render(
     <Button iconLeft={plusIcon} iconOnly={plusIcon} iconRight={plusIcon}>
       Button
     </Button>,
   );
 
-  expect(getAllByTestId('icon-only').length).toBe(1);
+  expect(screen.getAllByTestId('icon-only').length).toBe(1);
```
Comment (chanceaclark) [inline packages/big-design/src/components/Button/spec.tsx:187]:
```suggestion
  expect(screen.getAllByTestId('icon-only')).toHaveLength(1);
```

File: packages/big-design/src/components/Button/spec.tsx (hunk 1 above)
Comment (bc-alexsaiannyi) [reply packages/big-design/src/components/Button/spec.tsx:159]: Yeah, it's a good point. I also see that we will need @testing-library/user-event package for this purpose. Is it ok to proceed?

Comment (chanceaclark) [reply packages/big-design/src/components/Button/spec.tsx:159]: For sure

**Review by golcinho — APPROVED** [review]
Thank you 🙇

**Pull request merged**